packages/node_modules/@ciscospark/spark-core/src/lib/batcher.js

```diff
@@ -61,7 +61,10 @@ const Batcher = SparkPlugin.extend({
         this.prepareItem(item)
           .then((req) => {
             defer.promise = defer.promise
-              .then(() => this.deferreds.delete(idx))
+              .then((res) => {
+                this.deferreds.delete(idx);
+                return res;
+              })
```

Comment:
Returning the result of the predecessor chained promise instead of the value of delete.